### ISSUE TITLE
Issues from Interview 5

### ISSUE THREAD

**Issue by terracoda** [phetsims/balloons-and-static-electricity#124]
- [x] Need to clean-up some navigation cues: 1. First navigation cue in Scene Summary includes “Question mark for Keyboard Commands and Help”, but the Help dialog is only available when in forms mode, i.e. on the Balloon. Need to remove that Help nav cue from Scene Summary, until Help works from anywhere. 2. Navigation cue, after hitting Reset Balloons includes “Use WASD keys ...”. Need to remove.
- [x] Delayed reaction for the letter keys, likely due to performance on the Surface Pro 3.
- [x] User got stuff on Reset Button (literally stuck) (already noted on github).
- [x] Dialog: User can get back into content using the ARROW keys from the Dialog. Resources on this in Accessibility questions document. (#123)
- [x] JAWS stopped talking after exiting the Dialog, not sure why, but it took some doing to get it Talk again. I think this only happened once.

**Comment by jessegreenberg** [comment]
Just had a discussion with @terracoda.  The following change list was also suggested:

- [ ] Break up balloon description to restrict aria-live updated text
- [ ] Add document role to keyboard help dialog
- [ ] Add aria live region for wall addition and removal
- [ ] ResetAll button should be labelled as 'reset all'
- [ ] Ensure that the aria-live descriptions for the balloon and sweater are behaving correctly. Ideally, the balloon description change should be announced first, followed by the sweater.

**Comment by jessegreenberg** [comment]
> Need to clean-up some navigation cues: 1. First navigation cue in Scene Summary includes “Question mark for Keyboard Commands and Help”, but the Help dialog is only available when in forms mode, i.e. on the Balloon. Need to remove that Help nav cue from Scene Summary, until Help works from anywhere. 2. Navigation cue, after hitting Reset Balloons includes “Use WASD keys ...”. Need to remove.

@terracoda decided that this should stay as it is for the time being, navigation help will be filled in by the wizard for now.

**Comment by jessegreenberg** [comment]
New dev version published with the changes listed in https://github.com/phetsims/balloons-and-static-electricity/issues/124#issuecomment-178227911.

http://www.colorado.edu/physics/phet/dev/html/balloons-and-static-electricity/1.2.0-accessible-instance.11/balloons-and-static-electricity_en.html?accessibility

**Comment by terracoda** [comment]
@jessegreenberg Thanks for all of this!
Regarding the help dialog, I noticed with Instance 10 with JAWS, that without using the arrow keys, the dialog was cut short after the list item with the "Use A Key to move left." It was not the user that accidentally got back into the main content, but that the Dialog stopped and virtual focus ended up in an unexpected place.

I'll check Instance 11 with JAWS to see if it behaves better.

FYI, for instance 10 and NVDA, no content in the Dialog is read automatically and not knowing NVDA very well, I wasn't able to make it get read out.

**Comment by terracoda** [comment]
@jessegreenberg Regarding the Reset Button and version 10, I did get semi-stuck on it. I think it might have to do with the description and how the Accessibility Tree may be creating a nested structure from what we have in the DOM which is undesirable, but this is just a guess.
Behaviour:
I tab to the Reset Button, I get visual focus and the label text, then I tab again and visual focus goes away, but I keep getting "Reset Button" label text. I hit Shift Tab and I have to hit Shift Tab as many times as I hit Tab to get get all the way back to the Reset Balloons button. Basically, the Tabs get piled up instead wrapping to top, the browser bar. Again, I will check this in Instance 11.

**Comment by terracoda** [comment]
FYI, unfortunately, I am having serious computer performance issues which may be affecting the operation of the simulation. Rebooting to see if I get a fresh environment.

**Comment by jessegreenberg** [comment]
> It was not the user that accidentally got back into the main content, but that the Dialog stopped and virtual focus ended up in an unexpected place.

Ahh I see, thank you for explaining.  I will look into this.

> Regarding the Reset Button and version 10, I did get semi-stuck on it.

Very strange.  I will look into this as part of this issue.

> I am having serious computer performance issues which may be affecting the operation of the simulation. Rebooting to see if I get a fresh environment.

I experienced this as well when I was testing last night.  Seems unlikely, but there may aspects from the above commits that introduced some performance issues that are causing the AT to struggle.  I will investigate this.  High priority.

**Comment by terracoda** [comment]
@jessegreenberg With document role on Keyboard Commands and Help Dialog, I can read through line by line slowly! This is with JAWS and FireFox, and it seems to work as well with Internet Explorer, too!

**Comment by jessegreenberg** [comment]
@terracoda, clad to hear it is working with platform consistency!  :+1:

**Comment by jessegreenberg** [comment]
https://github.com/phetsims/balloons-and-static-electricity/issues/123 closed.

**Comment by jessegreenberg** [comment]
> JAWS stopped talking after exiting the Dialog, not sure why, but it took some doing to get it Talk again. I think this only happened once.

I could not reproduce this with JAWS, hopefully this has been fixed after recent changes to the dialog.

**Comment by terracoda** [comment]
@jessegreenberg I think JAWS stopped talking after the Dialog due to computer performance. I switched computers after my first interview and the remaining 6 interviews on a different computer went much more smoothly.

**Comment by jessegreenberg** [comment]
> Delayed reaction for the letter keys, likely due to performance on the Surface Pro 3.

Yes, this is a performance issue.  See #128.

**Comment by jessegreenberg** [comment]
> User got stuff on Reset Button (literally stuck) (already noted on github).

 See https://github.com/phetsims/balloons-and-static-electricity/issues/121

**Comment by jessegreenberg** [comment]
Closing this issue.  Issues either resolved or distributed into additional ```dev:a11y``` issues.
